### PR TITLE
Add some specs for Runner

### PR DESCRIPTION
**Pull request by JuanitoFatas**
I think these pending tests should belongs to the executor specs. Will do in following PRs.

- Test no Dangerfile present
- Test instance variables' default values
- Test default CLI options
- Test correctly sending messages to Executor
- Introduce new RSpec matcher: `have_instance_variables` (similar to have_attributes, but for instance variables)

### CODE REVIEW

File: spec/lib/danger/commands/runner_spec.rb [hunk 1]
```diff
@@ -1,126 +1,89 @@
...
+      runner = described_class.new(argv)
+      ui = runner.instance_variable_get(:"@cork")
+
+      expect(runner).to have_instance_variables(
+        "@dangerfile_path" => "Dangerfile",
+        "@base" => nil,
+        "@head" => nil,
+        "@fail_on_errors" => false,
+        "@danger_id" => "danger"
+      )
```
Comment (orta) [inline spec/lib/danger/commands/runner_spec.rb]: very cool

File: spec/lib/danger/commands/runner_spec.rb [hunk 2]
```diff
@@ -1,126 +1,89 @@
...
+        executor = double("Executor")
+
+        expect(Danger::Executor).to receive(:new) { executor }
+        expect(executor).to receive(:run).with(
+          base: "my-base",
+          head: "my-head",
+          dangerfile_path: "MyDangerfile",
+          danger_id: "my-danger-id",
+          fail_on_errors: "true"
+        )
```
Comment (orta) [inline spec/lib/danger/commands/runner_spec.rb]: ahhh, this is how rspec doubles work

File: spec/support/matchers/have_instance_variables_matcher.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  failure_message_when_negated do |actual|
+    expected.each do |instance_variable, expected_value|
+      actual_value = actual.instance_variable_get(instance_variable)
+      if actual_value == expected_value
+        return "expected #{actual}#{instance_variable} not to match #{expected_value}, but got #{actual_value}."
+      end
+    end
+  end
+end
```
Comment (orta) [inline spec/support/matchers/have_instance_variables_matcher.rb]: This is nice 👍

**Comment by orta** [comment]
I'm pumped about this matcher - slick work

**Comment by orta** [comment]
I feel guilty about all these squash and merges - I feel like you don't get as much credit in the commits

**Comment by JuanitoFatas** [comment]
> I feel guilty about all these squash and merges - I feel like you don't get as much credit in the commits

No worries, as long as danger keeps going 🎉

**Pull request merged**